src/mjolnir/landmark_builder.cc

```diff
@@ -1,10 +1,61 @@
-#include "mjolnir/landmark_database_builder.h"
+#include "mjolnir/landmark_builder.h"
 #include "filesystem.h"
+
+#include "mjolnir/osmpbfparser.h"
 #include "mjolnir/util.h"
+#include <boost/property_tree/ptree.hpp>
+#include <filesystem>
+#include <tuple>
+
+namespace {
+struct landmark_callback : public OSMPBF::Callback {
+public:
+  landmark_callback(const std::string& db_name) : db_(db_name, false) {
+  }
+  virtual ~landmark_callback() {
+  }
+
+  virtual void
+  node_callback(const uint64_t /*osmid*/, double lng, double lat, const OSMPBF::Tags& tags) override {
+    auto iter = tags.find("amenity");
+    if (iter != tags.cend() && !iter->second.empty()) {
+      valhalla::mjolnir::LandmarkType type = static_cast<valhalla::mjolnir::LandmarkType>(0);
+      try {
+        type = valhalla::mjolnir::string_to_landmark_type(iter->second);
+      } catch (const std::runtime_error& e) { return; }
+
+      std::string name = "";
+      auto it = tags.find("name");
+      if (it != tags.cend() && !it->second.empty()) {
+        name = it->second;
+      }
+
+      // insert parsed landmark directly into database
+      db_.insert_landmark(name, type, lng, lat);
```

Comment:
you can avoid this clunkiness by doing the whole thing inside the try catch. also dont use the name `type` as a variable name its just too close to keywords and likely other things it could collide with.

```suggestion
      try {
        auto landmark_type = valhalla::mjolnir::string_to_landmark_type(iter->second);
      
        std::string name = "";
        auto it = tags.find("name");
        if (it != tags.cend() && !it->second.empty()) {
          name = it->second;
        }

        // insert parsed landmark directly into database
        db_.insert_landmark(name, landmark_type, lng, lat);
      } catch (const std::runtime_error& e) { return; }
```

Reply:
nice suggestion thanks